src/features/SendAssets/SendAssets.reducer.tsx

```diff
@@ -0,0 +1,153 @@
+import { hexlify, bigNumberify, Arrayish } from 'ethers/utils';
+import { TransactionResponse } from 'ethers/providers';
+import { ValuesType } from 'utility-types';
+
+import {
+  IFormikFields,
+  ITxObject,
+  Asset,
+  TAddress,
+  ITxConfig,
+  ITxReceipt,
+  ISignedTx,
+  ITxHash,
+  ITxType,
+  ExtendedAsset,
+  Network,
+  StoreAccount,
+  TAction
+} from '@types';
+import {
+  getBaseAssetByNetwork,
+  hexWeiToString,
+  bigNumGasPriceToViewableGwei,
+  inputGasPriceToHex
+} from '@services';
+import { makePendingTxReceipt, makeTxConfigFromSignedTx } from '@utils';
+
+import { processFormDataToTx } from './helpers';
+
+interface State {
+  txConfig?: ITxConfig;
+  txReceipt?: ITxReceipt;
+  signedTx?: ISignedTx;
+  send?: boolean;
+}
+
+export type ReducerAction = TAction<ValuesType<typeof sendAssetsReducer.actionTypes>, any>;
+
+// @ts-ignore
+export const initialState: State = { txConfig: {} };
+
+export const sendAssetsReducer = (state: State, action: ReducerAction) => {
+  switch (action.type) {
+    case sendAssetsReducer.actionTypes.FORM_SUBMIT: {
+      const txConfig = handleFormSubmit(action.payload);
+      return { ...state, txConfig };
+    }
+    case sendAssetsReducer.actionTypes.SIGN: {
+      const { txConfig, signedTx } = handleSignedTx(state, action.payload);
+      return { ...state, txConfig, signedTx };
+    }
+    case sendAssetsReducer.actionTypes.WEB3_SIGN: {
+      const txReceipt = handlePendingTxReceipt(state, action.payload);
+      return { ...state, txReceipt };
+    }
+    case sendAssetsReducer.actionTypes.SEND: {
+      return { ...state, send: true };
+    }
+    case sendAssetsReducer.actionTypes.AFTER_SEND: {
+      const txReceipt = handleConfirmAndSend(state, action.payload);
+      return { ...state, txReceipt };
+    }
+    case sendAssetsReducer.actionTypes.RESUBMIT: {
+      const txConfig = handleResubmitTx(state);
+      return { ...state, txConfig };
+    }
+    case sendAssetsReducer.actionTypes.RESET:
+      return initialState;
+    default:
+      return state;
+  }
+};
+
+sendAssetsReducer.actionTypes = {
+  FORM_SUBMIT: 'FORM_SUBMIT',
+  SIGN: 'SIGN',
+  WEB3_SIGN: 'WEB3_SIGN',
+  SEND: 'SEND',
+  AFTER_SEND: 'AFTER_SEND',
+  RESUBMIT: 'RESUBMIT',
+  RESET: 'RESET'
+};
```

Comment:
Naming is hard. When it comes to actionTypes the name should reflect the action taken.
By convention I suggest we use `REQUEST, SUCCESS, FAILURE` to reflect the action that happened.
eg. https://github.com/MyCryptoHQ/MyCrypto/blob/285f33ed5be4c61c3c42fef91d60277853e62411/src/services/WalletService/walletconnect/reducer.ts
If if we don't use all three in this case, it will be easier to move some of the handling into the flow and expand in the future.

Reply:
I've tried renaming them slightly, let me know if it's clearer.